src/Native/System.Native/pal_io.cpp

```diff
@@ -443,3 +464,93 @@ int32_t MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     return mkstemps(pathTemplate, suffixLength);
 }
+
+static
+void ConvertPalPollFdToNative(PollFd& pal, pollfd& native)
+{
+    native.fd = pal.FileDescriptor;
+    native.events = pal.EventsToPollFor;
+    native.revents = pal.ReceivedEvents;
+}
+
+static
+void ConvertNativePollfdToPal(pollfd& native, PollFd& pal)
+{
+    pal.FileDescriptor = native.fd;
+    pal.EventsToPollFor = native.events;
+    pal.ReceivedEvents = native.revents;
+}
+
+extern "C"
+int32_t Poll(PollFd* pollData, uint32_t numberOfPollFds, int32_t timeout)
+{
+    // Convert our standardized pollfd to the native one
+    pollfd* fds = new pollfd[numberOfPollFds];
+    for (int i = 0; i < numberOfPollFds; i++)
+    {
+        ConvertPalPollFdToNative(pollData[i], fds[i]);
+    }
+
+    // Call poll with the native pollfd and, if necessary, convert the results back to our standardized pollfd
+    int32_t result = poll(fds, numberOfPollFds, timeout);
+    if (result > 0) // We only have result data if the result is positive
+    {
+        for (int i = 0; i < numberOfPollFds; i++)
+        {
+            ConvertNativePollfdToPal(fds[i], pollData[i]);
+        }
+    }
+
+    delete[] fds;
+
+    return result;
+}
+
+extern "C"
+int32_t Posix_FAdvise(int32_t fd, int64_t offset, int64_t length, AdviseAdvice advice)
+{
+#if HAVE_POSIX_ADVISE
+    return posix_fadvise(fd, offset, length, (int)advice);
+#else
+    // Not supported on this platform; however, we don't want to error out since
+    // currently, this isn't used on platforms where it isn't supported.
+    return ENOSYS;
+#endif
```

Comment:
Sorry for not being clear, I forgot the # before error (making it #error). Since we are looking for certain functionality, I wanted a comment to say that we explicitly aren't causing a compilation error here on purpose when this doesn't exist. I'll add the missing # on error to make that clear